### PR TITLE
Make storeRetained ordered using queues and bulk operations

### LINKED ISSUE #27: Not all packets are delivered to subscribed clients
Seems that there is a big bug with this peristence. When I publish 15 messages simultaniously I only get 5/6 of them in a client subscribed.

What could be this related to? I have tried to revert to some old versions but nothing changes. If I remove the persistence all messages are received from clients correctly so the problem must be here.

### PR DESCRIPTION
**Pull request by robertsLando**
This fixes #27 but I think it is a performance killer. @mcollina let me know what you think.

Whitout this PR connected clients will not receive all packets that is equally bad.

### CODE REVIEW

**Comment by robertsLando** [comment]
@mcollina Any thoughts ?

**Comment by robertsLando** [comment]
@mcollina Should be ok now

File: persistence.js [hunk 1]
```diff
@@ -15,6 +15,9 @@ var qlobberOpts = {
   wildcard_some: '#'
 }
 
+var packetsQueue = []
+var executing = false
```
Comment (mcollina) [inline persistence.js]: Can you move those two as fields in `MongoPersistence`?

**Comment by robertsLando** [comment]
@mcollina I think it should be added as a test in `abstract.js` of `aedes-cached-persistence`

**Comment by robertsLando** [comment]
Also I don't understand If you mean add the entire example as a separetd file or create a test based on that one. Anyway the best solution is to add it in the generic persistence repo in order to make sure all other persistences (and new one in future) will get that test

**Comment by mcollina** [comment]
We need a test based on that example.
Adding it to aedes-persistence is perfect.

**Comment by robertsLando** [comment]
@mcollina Ok, can you merge this pr and release an update? I will open an issue on aedes-persistence

**Comment by mcollina** [comment]
I would prefer to release once we have a test that proves that the bug is solved.

**Comment by robertsLando** [comment]
@mcollina done https://github.com/mcollina/aedes-persistence/pull/37

**Comment by mcollina** [comment]
that test fails here because of null/undefined issue, would you mind to fix that?

**Comment by robertsLando** [comment]
I’m not on my laptop right now, will try to find some time later or I will do it Monday sorry

Daniel

**Comment by robertsLando** [comment]
@mcollina Seems the problem was the test itself and not related to this. Check my new PR to aedes-persistence https://github.com/mcollina/aedes-persistence/pull/38

**Comment by robertsLando** [comment]
@mcollina Everything works on my side, seems like that travis is not taking aedes-persistence 7.2.1

**Comment by robertsLando** [comment]
@mcollina I confirm that trevis is using aedes-persistence 7.2.0 instead of latest 7.2.1 this is why tests are failing

**Comment by mcollina** [comment]
I had to clear travis caches

**Comment by robertsLando** [comment]
@mcollina Could you merge this now? Everything should be ok

**Pull request merged**